fMRIVolume/GenericfMRIVolumeProcessingPipeline.sh

```diff
@@ -293,15 +340,61 @@ if [ ! -e "$fMRIFolder" ] ; then
   log_Msg "mkdir ${fMRIFolder}"
   mkdir "$fMRIFolder"
 fi
-cp "$fMRITimeSeries" "$fMRIFolder"/"$OrigTCSName".nii.gz
+${FSLDIR}/bin/imcp "$fMRITimeSeries" "$fMRIFolder"/"$OrigTCSName"
+
+# --- Do slice time correction if indicated
+if [ $DoSliceTimeCorrection = "TRUE" ] ; then
+    log_Msg "Running slice timing correction using FSL's 'slicetimer' tool"
+    TR=`${FSLDIR}/bin/fslval ${fMRIName} pixdim4`
+
+    IFS='@' read -a SliceTimerCorrectionParametersArray <<< "$SliceTimerCorrectionParameters"
+    ${FSLDIR}/bin/immv "$fMRIFolder"/"$OrigTCSName" "$fMRIFolder"/"$OrigTCSName"_prestc
+    ${FSLDIR}/bin/slicetimer -i "$fMRIFolder"/"$OrigTCSName"_prestc -o "$fMRIFolder"/"$OrigTCSName" -r ${TR} -v "${SliceTimerCorrectionParametersArray[@]}"
+    ${FSLDIR}/bin/imrm "$fMRIFolder"/"$OrigTCSName"_prestc
+fi
 
 #Create fake "Scout" if it doesn't exist
 if [ $fMRIScout = "NONE" ] ; then
   ${RUN} ${FSLDIR}/bin/fslroi "$fMRIFolder"/"$OrigTCSName" "$fMRIFolder"/"$OrigScoutName" 0 1
 else
-  cp "$fMRIScout" "$fMRIFolder"/"$OrigScoutName".nii.gz
+  ${FSLDIR}/bin/imcp "$fMRIScout" "$fMRIFolder"/"$OrigScoutName"
 fi
 
+if [ $DistortionCorrection = "NONE" ] ; then
+    # Processing is more robust to registration problems if the fMRI is in the same orientation as the
+    # standard template (MNI152) images, which can be accomplished using FSL's `fslreorient2std`.
+    # HOWEVER, if you reorient, other parameters (such as UnwarpDir) need to be adjusted accordingly.
+    # Rather than deal with those complications here, we limit reorienting to DistortionCorrection=NONE condition.
+    xorient=`$FSLDIR/bin/fslval "$fMRIFolder"/"$OrigTCSName" qform_xorient | tr -d ' '`
```

Comment:
Add newline, then following comment
`#First though, detect if reorienting is even necessary`

Reply:
Added comment.